scripts/build_test.py

```diff
@@ -721,4 +722,109 @@ def mock_require_compiled_js_dir_to_be_valid():
                 self.assertFalse(
                     os.path.exists(os.path.dirname(MOCK_COMPILED_JS_DIR)))
 
+    def test_build_with_prod_env(self):
+        check_function_calls = {
+            'build_using_webpack_gets_called': False,
+            'ensure_files_exist_gets_called': False,
+            'compile_typescript_files_gets_called': False
+        }
+        expected_check_function_calls = {
+            'build_using_webpack_gets_called': True,
+            'ensure_files_exist_gets_called': True,
+            'compile_typescript_files_gets_called': True
+        }
+
+        def mock_build_using_webpack():
+            check_function_calls['build_using_webpack_gets_called'] = True
+
+        def mock_ensure_files_exist(unused_filepaths):
+            check_function_calls['ensure_files_exist_gets_called'] = True
+
+        def mock_compile_typescript_files(unused_project_dir):
+            check_function_calls['compile_typescript_files_gets_called'] = True
+
+        ensure_files_exist_swap = self.swap(
+            build, '_ensure_files_exist', mock_ensure_files_exist)
+        build_using_webpack_swap = self.swap(
+            build, 'build_using_webpack', mock_build_using_webpack)
+        compile_typescript_files_swap = self.swap(
+            build, 'compile_typescript_files', mock_compile_typescript_files)
+        args_swap = self.swap(sys, 'argv', ['build.py', '--prod_env'])
+
+        with ensure_files_exist_swap, build_using_webpack_swap, (
+            compile_typescript_files_swap), args_swap:
+            build.build()
+
+        self.assertEqual(check_function_calls, expected_check_function_calls)
+
+    def test_build_with_watcher(self):
+        check_function_calls = {
+            'ensure_files_exist_gets_called': False,
+            'compile_typescript_files_gets_called': False
+        }
+        expected_check_function_calls = {
+            'ensure_files_exist_gets_called': True,
+            'compile_typescript_files_gets_called': True
+        }
+
+        def mock_ensure_files_exist(unused_filepaths):
+            check_function_calls['ensure_files_exist_gets_called'] = True
+
+        def mock_compile_typescript_files(unused_project_dir):
+            check_function_calls['compile_typescript_files_gets_called'] = True
+
+        ensure_files_exist_swap = self.swap(
+            build, '_ensure_files_exist', mock_ensure_files_exist)
+        compile_typescript_files_swap = self.swap(
+            build, 'compile_typescript_files_continuously',
+            mock_compile_typescript_files)
```

Comment:
Also, here and elsewhere, consider also adding checks to ensure that particular functions are _not_ called when they are not supposed to be.

Reply:
Then it will decrease coverage of build_test file as those mock functions would never be called.

Reply:
What if you use test_utils.CallCounter, does that help?

Also, is there a way to omit coverage of test files from the report? We don't care about coverage of test files -- we care about coverage of the main codebase.